### PR TITLE
Simplify cache and remove TempestHyperdiffusion

### PR DESCRIPTION
**Pull request by dennisYatunin**
## Purpose 

This PR is also split off from #1512. It removes `TempestHyperdiffusion`, which is no longer needed, and slightly simplifies the cache `p`.

## Content

- Remove the `TempestHyperdiffusion` struct and the `enable_qt_hyperdiffusion` function. Remove the `B` parameter from the `ClimaHyperdiffusion` struct.
- Allow the `hyperdiff` CLI option to be `ClimaHyperdiffusion`, `true`, `none` or `false`, and remove the `enable_qt_hyperdiffusion` CLI option.
- Rename `horizontal_limiter_tendency!` to `limited_tendency!`, and rename `hyperdiffusion_tracers_tendency!` to `tracer_hyperdiffusion_tendency!`.
- Move the `hyperdiffusion_cache` to `default_cache`, and move the `hyperdiffusion_tendency!` to `remaining_tendency!`.
- Move the `NVTX.@range` outside of `hyperdiffusion_tendency!` and `tracer_hyperdiffusion_tendency!`.
- Separate the `hyperdiffusion_ghost_buffer` from the main `ghost_buffer`, and allocate it in `hyperdiffusion_cache`.
- Replace `p.ghost_buffer.skip_dss` with `p.do_dss`.
- Only apply DSS during hyperdiffusion if `do_dss` is `true`.
- Replace `ᶜρh` with `p.ᶜtemp_scalar`, `ᶜω³` with `p.ᶜtemp_CT3`, and `ᶠω¹²` with `p.ᶠtemp_CT12`.
- Replace the n identical copies of the limiter in `p` with a single limiter.

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: perf/jet_nfailures.jl [hunk 1]
```diff
@@ -35,7 +35,7 @@ using Test
     # inference. By increasing this counter, we acknowledge that
     # we have introduced an inference failure. We hope to drive
     # this number down to 0.
-    n_allowed_failures = 30
+    n_allowed_failures = 31
```
Comment (charleskawczynski) [inline perf/jet_nfailures.jl:38]: What’s the inference failure being introduced? Can you try to fix this if it’s not too much trouble?

**Review by charleskawczynski — COMMENTED** [review]
Can we fix the inference failure?

and what is behavior changing in this PR?

**Comment by dennisYatunin** [comment]
@charleskawczynski The behavior change is not really a behavior change--I moved the hyperdiffusion tendency from the additional tendency to right after the horizontal advection tendency, and this change in the order of floating point additions modified the last bit or two in the remaining tendency. I'll see if I can fix the inference failure.

**Comment by dennisYatunin** [comment]
@charleskawczynski The inference failure is from moving the `NVTX.@range` that was in `hyperdiffusion_tendency!` to `remaining_tendency!` and changing it to an `@nvtx`. I think we can let this slide for now, right?

**Comment by charleskawczynski** [comment]
Sounds good, thank you @dennisYatunin!

**Pull request merged**